### PR TITLE
Engine fixes

### LINKED ISSUE #21215: hybrid power source vehicles can not be controlled remotely
vehicles with manual power (i..e, foot pedals) and real power (i.e., minireactor) cannot be controlled remotely, or even from an adjacent tile to turn on the engine while the manual engine is still on.

currently one has to, in order, take control of the vehicle (fail to start the "engines"), change the engine configuration, and then start the engines

the manual engines should automagically turn off when controlling a vehicle remotely (or be ignored)(and turn back on/reset to its previous state when the remote controlling stops)

### LINKED ISSUE #26793: Negitive sound error when leaving electric bike.
**Describe the bug**  
Getting error with negitive sound with sound.cpp ( see attached screenshot ) 

**To Reproduce**  
Unsure if getting on a electric bike and then dismounting into grass causes this problem but it was what did it for me. 

**Expected behavior**  
No error and correct sound level. 

**Screenshots**  
image

**Versions and configuration(please complete the following information):**
 - OS: Arch Linux 64bit 
 - Game Version: 0.C-34526-gae643fe3c6
 - Graphics version: Terminal
 - Mods loaded: Filthy Clothing, Boats, Vehicle Addtions Pack, Tanks and other., Beta National guard campt, Mutant NPCs , Custom Personal mod ( recipes only fixes, spoil timers etc, json only no LUA code ) 

**Additional context**  
[Spaulding.zip](https://github.com/CleverRaven/Cataclysm-DDA/files/2605229/Spaulding.zip)

### PR DESCRIPTION
**Pull request by LetterShapedGlyphs**
#### Summary
```SUMMARY: Bugfixes "Fixes errors in engine starting and idling"```

#### Purpose of change
Fixes #26793 (Muscle engines could idle and emit negative sound if paired with a disabled second engine. Also incorporates additional fixes/changes recommended by mlangsdorf.)
Fixes #21215 (Vehicles with multiple engine types did not start properly)

#### Describe the solution
Fixed the fuel checks for muscle engines to require a character with unbroken (relevant) limbs to be on the tile with the engine. Fixed the engine power calculation to only count engines that could actually run. Fixed the premature termination of engine startup when one failed.

#### Describe alternatives you've considered
N/A

#### Additional context
This also prevents players from counting as a muscle power source when the limbs they would be using are broken, and allows for vehicles to start as long as at least one of the enabled engines successfully starts.

### CODE REVIEW

File: src/activity_handlers.cpp [hunk 1]
```diff
@@ -2104,33 +2104,45 @@ void activity_handlers::start_engines_finish( player_activity *act, player *p )
     }
 
     int attempted = 0;
+    int non_muscle_attempted = 0;
```
[the remark this answers is not in the document]
Comment (Karthas077) [reply src/activity_handlers.cpp]: That's how it was before, but it makes the following checks involving non-muscle engines require a lot of subtracting and was more math and more confusing.

[the remark this answers is not in the document]
Comment (mlangsdorf) [reply src/activity_handlers.cpp]: I see your point.  The summing would be clearer, but the if logic worse.  Request withdraw.

File: src/vehicle.cpp [hunk 2]
```diff
@@ -2724,7 +2730,7 @@ int vehicle::total_power_w( bool const fueled, bool const safe ) const
     for( size_t e = 0; e < engines.size(); e++ ) {
         int p = engines[e];
         if( is_engine_on( e ) &&
-            ( !fueled || is_perpetual_type( e ) || fuel_left( part_info( p ).fuel_type ) ) ) {
+            ( !fueled || fuel_left( part_info( p ).fuel_type ) ) ) {
```
[the remark this answers is not in the document]
Comment (Karthas077) [reply src/vehicle.cpp]: Vortex engines still start and produce thrust for vehicles when installed.

**Comment by Tsunder** [comment]
thanks!

**Pull request merged**